src/lib/comms/sol-lwm2m-client.c

```diff
@@ -52,6 +53,117 @@ SOL_LOG_INTERNAL_DECLARE_STATIC(_lwm2m_client_domain, "lwm2m-client");
+    do { \
+        r = sol_coap_server_unregister_resource(client->coap_server, \
+            __resource); \
+        SOL_INT_CHECK(r, < 0, r); \
+ \
+        if (client->security) { \
+            if (sol_lwm2m_security_supports_security_mode(client->security, \
+                SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY)) { \
+                r = sol_coap_server_unregister_resource(client->dtls_server_psk, \
+                    __resource); \
+                SOL_INT_CHECK(r, < 0, r); \
+            } \
+ \
+            if (sol_lwm2m_security_supports_security_mode(client->security, \
+                SOL_LWM2M_SECURITY_MODE_RAW_PUBLIC_KEY)) { \
+                r = sol_coap_server_unregister_resource(client->dtls_server_rpk, \
+                    __resource); \
+                SOL_INT_CHECK(r, < 0, r); \
+            } \
+        } \
+    } while (0);
+
+#define SOL_COAP_SERVER_REGISTER_RESOURCE_ALL_GOTO(__resource, \
+        __err_label_1, __err_label_2, __err_label_3) \
+    do { \
+        r = sol_coap_server_register_resource(client->coap_server, \
+            __resource, client); \
+        SOL_INT_CHECK_GOTO(r, < 0, __err_label_1); \
+ \
+        if (client->security) { \
+            if (sol_lwm2m_security_supports_security_mode(client->security, \
+                SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY)) { \
+                r = sol_coap_server_register_resource(client->dtls_server_psk, \
+                    __resource, client); \
+                SOL_INT_CHECK_GOTO(r, < 0, __err_label_2); \
+            } \
+ \
+            if (sol_lwm2m_security_supports_security_mode(client->security, \
+                SOL_LWM2M_SECURITY_MODE_RAW_PUBLIC_KEY)) { \
+                r = sol_coap_server_register_resource(client->dtls_server_rpk, \
+                    __resource, client); \
+                SOL_INT_CHECK_GOTO(r, < 0, __err_label_3); \
+            } \
+        } \
+    } while (0);
+
+#define SOL_COAP_SERVER_REGISTER_RESOURCE_ALL_INT(__resource) \
+    do { \
+        r = sol_coap_server_register_resource(client->coap_server, \
+            __resource, client); \
+        SOL_INT_CHECK(r, < 0, r); \
+ \
+        if (client->security) { \
+            if (sol_lwm2m_security_supports_security_mode(client->security, \
+                SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY)) { \
+                r = sol_coap_server_register_resource(client->dtls_server_psk, \
+                    __resource, client); \
+                SOL_INT_CHECK(r, < 0, r); \
+            } \
+ \
+            if (sol_lwm2m_security_supports_security_mode(client->security, \
+                SOL_LWM2M_SECURITY_MODE_RAW_PUBLIC_KEY)) { \
+                r = sol_coap_server_register_resource(client->dtls_server_rpk, \
+                    __resource, client); \
+                SOL_INT_CHECK(r, < 0, r); \
+            } \
+        } \
+    } while (0);
+
+#define SOL_COAP_NOTIFY_BY_CALLBACK_ALL_INT(__resource) \
```

Comment:
The names were not based on the return value, but on the kind of `SOL_*_CHECK[_*]` used.